packages/labs/context/src/lib/controllers/context-provider.ts

```diff
@@ -56,7 +56,7 @@ export class ContextProvider<T extends ContextKey<unknown, unknown>>
   public onContextRequest = (
     ev: ContextRequestEvent<ContextKey<unknown, unknown>>
   ): void => {
-    if (ev.context !== this.context) {
+    if (ev.context !== this.context || ev.composedPath()[0] === this.host) {
```

Comment:
Can you add a comment here on why we do this check?

Reply:
resolved by [f85f2d9](https://github.com/lit/lit/pull/2858/commits/f85f2d985300b97b0a531652613e413482679df9)